src-tauri/src/tray/updater.rs

```diff
@@ -12,29 +12,56 @@ pub fn check_for_update(app_handle: AppHandle) -> Result<()> {
                                 .ok_button_label("Install")
                                 .cancel_button_label("Cancel")
-                                .show(|result| match result {
-                                    true => {
-                                        // TODO: install update
-                                        // update.download_and_install(None, None);
-                                    }
-                                    _ => {}
+                                .show(move |result| {
+                                    tauri::async_runtime::spawn(async move {
+                                        match result {
+                                            true => {
+                                                let update_result = update
+                                                    .clone()
+                                                    .download_and_install(
+                                                        |size, _| {
+                                                            println!("downloading update...");
+                                                            println!("size: {}", size);
+                                                        },
+                                                        || {
+                                                            println!("update downloaded, proceeding to install!");
+                                                        },
+                                                    )
+                                                    .await;
+
+                                                match update_result {
+                                                    Ok(_) => {
+                                                        app_handle
+                                                            .dialog()
+                                                            .message("Update installed successfully!")
+                                                            .ok_button_label("Relaunch")
+                                                            .show(move |_| {
+                                                                app_handle.restart();
+                                                            });
+                                                    }
+                                                    Err(e) => {
+                                                        // TODO: handle a failed update
+                                                        eprintln!("failed to install update: {}", e);
+                                                    }
+                                                    
+                                                }
+                                            }
+                                            _ => {}
+                                        }
+                                    });
```

Comment:
Hmm, wondering why not just do a simple

```
 "check_updates" => {
                    let w = app.get_window("main").unwrap();
                    w.trigger_global("tauri://update", None);
                }    
```

and let Tauri handle the rest?  @clearlysid

Reply:
The above is what I did for [yack](https://github.com/rajatkulkarni95/yack/blob/main/src-tauri/src/main.rs)

which is also a menu bar app. Doesn't the current approach seem too verbose?
<img width="278" alt="Screenshot 2024-04-28 at 21 48 37" src="https://github.com/helmerapp/micro/assets/57321156/c90de74d-0445-42f0-9b9b-4c8efc6def56">

Reply:
Thanks for the suggestion dude! I didn't know about the `tauri://update` 😅 Looks so much cleaner. Will reopen this PR and give it a try.